presentation/components/src/presentation-components/table/DataProvider.ts

```diff
@@ -256,9 +257,57 @@ const createColumns = (descriptor: Readonly<Descriptor> | undefined): ColumnDesc
     return [createLabelColumn()];
 
   const sortedFields = [...descriptor.fields].sort(priorityAndNameSortFunction);
+
+  extractSameInstanceFields(sortedFields);
+
   return sortedFields.map((field) => createColumn(field));
 };
 
+const extractSameInstanceFields = (fields: Field[]) => {
+  const foundFields: { [field: string]: Field } = {};
+  for (let i = 0; i < fields.length; i++) {
+    const field = fields[i];
+    if (field.isNestedContentField() && field.relationshipMeaning === RelationshipMeaning.SameInstance) {
+      const nestedFields = field.nestedFields.map((nestedField: Field): Field => (
+        new Field(

```

Comment:
Do you really need a copy here? If yes, you should use `clone` function for that.

Reply:
I just needed to reset parentship. Updated

Reply:
Why do you need to reset parentship? That mutates the descriptor which we don't want.